packages/react-storage/src/components/StorageBrowser/composables/DataTable/dataCells/ButtonDataCell.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import {
+  ButtonElement,
+  ButtonElementVariant,
+  IconElement,
+  IconVariant,
+} from '../../../context/elements';
+import { CLASS_BASE } from '../../../views/constants';
+
+export interface ButtonDataCellProps {
+  content: {
+    icon?: IconVariant;
+    label?: string;
+    onClick?: () => void;
+  };
+}
+
+export const ButtonDataCell = ({
+  content,
+}: ButtonDataCellProps): React.JSX.Element => {
+  const { icon, label, onClick } = content;
+
+  // Special handling for icon-only cancel buttons
+  let buttonVariant: ButtonElementVariant = 'table-data';
+  if (icon === 'cancel' && !label) {
+    buttonVariant = 'cancel';
+  }
```

Comment:
Might make sense to add an icon button cell since we also need to consider remove, download, delete, etc